src/dmd/traits.d

```diff
@@ -911,6 +912,51 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         }
         return symbolToExp(s, e.loc, sc, false);
     }
+    if (e.ident == Id.child)
+    {
+        if (dim != 2)
+            return dimError(2);
+
+        Expression ex;
+        auto op = (*e.args)[0];
+        if (auto symp = getDsymbol(op))
+            ex = new DsymbolExp(e.loc, symp);
+        else
+        {
+            ex = op.isExpression();
+            if (!ex)
+            {
+                e.error("symbol or expression expected as first argument of __traits `%s` instead of `%s`",
+                        e.ident.toChars(), op.toChars());
+                return new ErrorExp();
+            }
+        }
```

Comment:
```suggestion
        else if (auto exp = op.isExpression())
            ex = exp;
        else
        {
            e.error("symbol or expression expected as first argument of __traits `%s` instead of `%s`", e.ident.toChars(), op.toChars());
            return new ErrorExp();
        }
```